src/aws-api-mcp-server/awslabs/aws_api_mcp_server/core/common/cloudwatch_logger.py

```diff
@@ -0,0 +1,152 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import boto3
+import threading
+import time
+from ..aws.credentials import get_local_credentials
+from ..common.config import AWS_API_MCP_PROFILE_NAME
+from botocore.exceptions import ClientError
+from loguru import logger
+from typing import Optional
+
+
+class CloudWatchLogSink:
```

Comment:
Is the log pushing done in a separate thread than the main process thread that reacts to MCP requests? CloudWatch can be down, pushing logs can be slow etc. Have we considered if this should be pushed in a separate thread to avoid blocking answering MCP requests instead?

Reply:
it is done in a separate thread